src/main/java/macrobase/analysis/outlier/OutlierClassifier.java

```diff
@@ -0,0 +1,19 @@
+package macrobase.analysis.outlier;
+
+import macrobase.analysis.result.OutlierClassificationResult;
+import macrobase.conf.MacroBaseConf;
+import macrobase.datamodel.Datum;
+
+public abstract class OutlierClassifier {
+
+    private OutlierClassificationResult lastResult;
+
+    public OutlierClassifier(MacroBaseConf conf) {}
+    public abstract void observe(Datum datumToClassify);
+
+    public abstract OutlierClassificationResult classify();
```

Comment:
So what is the difference between `classify` and `getResult`? And if it classifies all of the data does that mean it has to keep the whole input stream around? That seems like a bad idea for high volume streams.